lib/matplotlib/testing/decorators.py

```diff
@@ -218,8 +244,12 @@ def compare(self, idx, baseline, extension, *, _lock=False):
                 if _lock else contextlib.nullcontext())
         with lock:
             fig.savefig(actual_path, **kwargs)
-            expected_path = self.copy_baseline(baseline, extension)
-            _raise_on_image_difference(expected_path, actual_path, self.tol)
+            if generating:
+                expected_creation_path = self.create_baseline(baseline, extension)
```

Comment:
I don't think we need to capture this variable?

Reply:
Done